spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ConvLSTMPeephole.scala

```diff
@@ -110,8 +116,12 @@ class ConvLSTMPeephole[T : ClassTag](
           .add(h2g))
     }
 
+    // make a new instance of innerActivation for each build gate
+    val innerActivation2 = innerActivation.cloneModule()
+    innerActivation2.setName(Integer.toHexString(java.util.UUID.randomUUID().hashCode()))
```

Comment:
This may impact the performance and why we need to re-create this every time?

Reply:
Actually in the original code, it uses .add(Sigmoid()), it is creating a new instance every time. Here I just clone the input activation to make a new instance.
I think that's basically the same?

Reply:
Yes, we need create a new sigmoid every time buildGate is called. I think we can try to move "if (innerActivation == null) innerActivation = Sigmoid[T]()" into buildGate, change innerActivation from a class member to a local variable, it will make sure the logic is the same and will not impact the performance.